tests/feature_selection/test_relevance.py

```diff
@@ -0,0 +1,133 @@
+        assert 'classification' == infer_ml_task(y)
+
+    def test_infers_regression_for_float_target(self):
+        y = pd.Series([1.0, 1.5, 1.7])
+        assert 'regression' == infer_ml_task(y)
+
+
+class TestCalculateRelevanceTable:
+    @pytest.fixture()
+    def y_binary(self):
+        return pd.Series([0, 1, 1])
+
+    @pytest.fixture()
+    def y_real(self):
+        return pd.Series([0.1, 0.2, 0.1])
+
+    @pytest.fixture()
+    def X(self):
+        df = pd.DataFrame()
+        df['feature_binary'] = [1, 1, 0]
+        df['feature_real'] = [0.1, 0.2, 0.3]
+        return df
+
+    def test_restrict_ml_task_options(self, X, y_binary):
+        with pytest.raises(ValueError):
+            calculate_relevance_table(X, y_binary, ml_task='some_other_task')
+
+    def test_constant_feature_irrelevant(self, y_binary):
+        X = pd.DataFrame([1, 1, 1], columns=['feature_binary'])
+
+        relevance_table = calculate_relevance_table(X, y_binary)
+        assert "feature_binary" == relevance_table.index[0]
+        assert 'constant' == relevance_table.type[0]
+        assert np.isnan(relevance_table.p_value[0])
+        assert False == relevance_table.relevant[0]
+
+    @mock.patch('tsfresh.feature_selection.relevance.target_binary_feature_real_test')
+    @mock.patch('tsfresh.feature_selection.relevance.target_binary_feature_binary_test')
+    def test_target_binary_calls_correct_tests(self, significance_test_feature_binary_mock,
+                                         significance_test_feature_real_mock, X, y_binary):
+        significance_test_feature_binary_mock.return_value = 0.5
+        significance_test_feature_real_mock.return_value = 0.7
+        relevance_table = calculate_relevance_table(X, y_binary, n_jobs=0)
+
+        assert 0.5 == relevance_table.loc['feature_binary'].p_value
+        assert 0.7 == relevance_table.loc['feature_real'].p_value
+        assert 2 == significance_test_feature_binary_mock.call_count
+        assert 2 == significance_test_feature_real_mock.call_count
+
+    @mock.patch('tsfresh.feature_selection.relevance.target_real_feature_real_test')
+    @mock.patch('tsfresh.feature_selection.relevance.target_real_feature_binary_test')
+    def test_target_real_calls_correct_tests(self, significance_test_feature_binary_mock,
+                                         significance_test_feature_real_mock, X, y_real):
+        significance_test_feature_binary_mock.return_value = 0.5
+        significance_test_feature_real_mock.return_value = 0.7
+
+        relevance_table = calculate_relevance_table(X, y_real, n_jobs=0)
+
+        assert 0.5 == relevance_table.loc['feature_binary'].p_value
+        assert 0.7 == relevance_table.loc['feature_real'].p_value
+        significance_test_feature_binary_mock.assert_called_once_with(X['feature_binary'], y=y_real)
+        significance_test_feature_real_mock.assert_called_once_with(X['feature_real'], y=y_real)
+
+
+class TestCombineRelevanceTables:
```

Comment:
don't you want to test if the relevance_tables are combined? so you need two relevance tables with different indices?